lib/postgrex/protocol.ex

```diff
@@ -3185,9 +3187,13 @@ defmodule Postgrex.Protocol do
   end
 
   defp columns(fields) do
-    fields
-    |> Enum.map(fn row_field(type_oid: oid, name: name) -> {oid, name} end)
-    |> :lists.unzip()
+    {oids_mods, names} =
+      fields
+      |> Enum.map(fn row_field(type_oid: oid, type_mod: mod, name: name) -> {{oid, mod}, name} end)
+      |> :lists.unzip()
+
+    {oids, mods} = :lists.unzip(oids_mods)
```

Comment:
You can keep it as a flat list and use unzip3: https://www.erlang.org/doc/apps/stdlib/lists.html#unzip3/1